### PR TITLE
Alternative solution to enable PHP unit tests for library classes using facades

### PR DESCRIPTION
**Pull request by pikanji**
This is alternative solution to #296.

Changes

- Add dependency to [orchestra/testbench](https://github.com/orchestral/testbench)
  orchestra/testbench enables unit tests for the classes using Laravel features such as facades and helper functions. This is the library used in laravel/framework to test such classes.
- Update to PHP Unit 6, because orchestra/testbench requires.
- Fix TestCase.php for PHP Unit 6.
- Add tests/Database/Attach/FileTest.php
  Sample test class to demonstrate test for class using facades.
- Fix tests/Halcyon/ModelTest.php
  The subject class of this test uses facade, and the test fails if an application is initialized even in other test. It is fixed by using orchestra/testbench TestCase class and properly calling parent's setUp method.
- Fix src/Database/Attach/File.php for local root path
  The local root path is configurable, and thus, configured value should be returned.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "laravel/tinker": "~1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~5.7",
+        "phpunit/phpunit": "~6.5",
```
[the remark this answers is not in the document]
Comment (pikanji) [reply composer.json]: Thanks for update!

[the remark this answers is not in the document]
Comment (pikanji) [reply composer.json]: haha, I had the same experience!

[the remark this answers is not in the document]
Comment (pikanji) [reply composer.json]: I tried to remove this requirement of upgrading phpunit to 6 by changing version of orchestra/testbench to the one compatible with phpunit 5.x.
But, that version of orchestra/testbench is not compatible with Laravel 5.5+. So, the requirement of phpunit 6 stays.

[the remark this answers is not in the document]
Comment (LukeTowers) [reply composer.json]: No longer an issue, octobercms/october is requiring 6.5 now too

File: src/Database/Attach/File.php [hunk 2]
```diff
@@ -976,6 +977,6 @@ protected function getPartitionDirectory()
      */
     protected function getLocalRootPath()
     {
-        return storage_path().'/app';
+        return Config::get('filesystems.disks.local.root');
```
Comment (bennothommo) [inline src/Database/Attach/File.php]: @pikanji This change is not necessary - the `getLocalRootPath` method is *only* called when the storage in use is "local".

**Comment by bennothommo** [comment]
@pikanji Looks pretty nifty - Testbench looks very useful. This will need to be tested against our main October repository tests as well, however, to ensure there's no conflicts that arise from its use.

**Comment by bennothommo** [comment]
Closing as it has been over a month since any activity on this occurred and we are trying to figure out what issues are still relevant. If this is still something that you would like to see through to fruition please respond and we can get the ball rolling.

**Pull request closed without merging**